cli/internal/taskhash/taskhash.go

```diff
@@ -277,18 +277,68 @@ func (th *Tracker) CalculateFileHashes(
 	return nil
 }
 
-type taskHashInputs struct {
+type taskHashable struct {
 	packageDir           turbopath.AnchoredUnixPath
 	hashOfFiles          string
 	externalDepsHash     string
 	task                 string
 	outputs              fs.TaskOutputs
 	passThruArgs         []string
+	envMode              util.EnvMode
+	passthroughEnv       []string
 	hashableEnvPairs     []string
 	globalHash           string
 	taskDependencyHashes []string
 }
 
+type oldTaskHashable struct {
+	packageDir           turbopath.AnchoredUnixPath
+	hashOfFiles          string
+	externalDepsHash     string
+	task                 string
+	outputs              fs.TaskOutputs
+	passThruArgs         []string
+	hashableEnvPairs     []string
+	globalHash           string
+	taskDependencyHashes []string
+}
+
+// calculateTaskHashFromHashable returns a hash string from the taskHashable
+func calculateTaskHashFromHashable(full *taskHashable) (string, error) {
+	switch full.envMode {
+	case util.Infer:
+		if full.passthroughEnv != nil {
+			// In infer mode, if there is any passThru config (even if it is an empty array)
+			// we'll hash the whole object, so we can detect changes to that config
+			// Further, resolve the envMode to the concrete value.
+			full.envMode = util.Strict
```

Comment:
I see that we're "resolving" the mode to strict. Will this show up in Run Summary as `strict` or `infer`? We should do the same behavior in both places, or add `envMode` and `resolvedEnvMode`. Don't have a strong preference, but this is leading me to thing we need a `runOptions` key in Run Summary that shows the raw values typed by the user, and then the rest of the data is resolved.

Reply:
Global: can present infer (meaning per-task), strict, or loose.
Task: can present only strict or loose.

I'll double check to make sure this is correct and there are tests for it.

Reply:
The updated commit ensures that this invariant is respected.